### PR TITLE
feat: [COMP-3486]: add security step to lite engine

### PR DESCRIPTION
**Pull request by r0610205**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30392)

### CODE REVIEW

**Comment by r0610205** [comment]
trigger messagecheck

**Comment by r0610205** [comment]
trigger messagecheck

File: product/ci/engine/proto/execution.proto [hunk 1]
```diff
@@ -97,6 +110,7 @@ message UnitStep {
     PublishArtifactsStep publish_artifacts = 6;
     PluginStep plugin = 9;
     RunTestsStep runTests = 10;
+    SecurityStep security = 15;
```
Comment (shubham149) [inline product/ci/engine/proto/execution.proto:113]: How is security step different from plugin step? If not, we should use plugin step only. The same happens for git OOTB step as well as docker OOTB step

Comment (r0610205) [reply product/ci/engine/proto/execution.proto:113]: It's actually different, because we need to grab env var outputs

Comment (shubham149) [reply product/ci/engine/proto/execution.proto:113]: can you point me to the code that handles this?

**Pull request closed without merging**